Fraction output

bmg1919 opened the pull request:
This may need further testing. All fraction output is mixed if numerator is larger than denominator. I can add an option for improper fractions if desired, maybe even a button to switch the current result from one to the next (mixed/improper, proper, decimal).

Darkempire78 commented:
Thanks for your PR, here is some changes to do before merging:
- [ ] Add icons to both setting option
- [ ] When you click on then result, it should display the default result (and vice versa)
- [ ] Should not display the fraction output if there is no calculation
- [ ] The app crashes when you put a big number like "5555555555555555" (no idea of why)
- [ ] Perhaps change the output format to an html format like "<sup>$num</sup>/<sub>$deno</sub>", because it's very not beautiful right now

Darkempire78 commented:
any news?

bmg1919 commented:
> any news?

Other than icons, I have some changes near ready to push.

bmg1919 commented:
> Thanks for your PR, here is some changes to do before merging:
> 
> * [ ]  Add an icon to both setting options (the first one is already used by another option)
> * [ ]  When you click on then result, it should display the default result (and vice versa)
> * [*]  Should not display the fraction output if there is no calculation
> * [*]  The app crashes when you put a big number like "5555555555555555" (no idea of why)
> * [*]  Perhaps change the output format to an html format like "$num/$deno", because it's very not beautiful right now

I've directly addressed the issues I've marked. I need to find or make some different icons for the menu option. I did reuse the precision icon. Instead of touching the screen to switch results, it currently shows both decimal and fraction results on pressing the equals button.

clementwzk commented:
any news? @bmg1919

clementwzk commented:
i will check that now, can you solve the conflicts btw ?

clementwzk commented:
<img width="544" height="217" alt="image" src="https://github.com/user-attachments/assets/bdbe6d4b-f466-4b5d-ad50-e67645dc4746" />
i do not really inderstand why i cannot use this branch. I do no see another one with the same name, it's very weird

bmg1919 commented:
Working on it
